mne/io/fiff/tests/test_raw.py

```diff
@@ -115,6 +115,33 @@ def test_copy_append():
     assert_equal(data.shape[1], 2 * raw._data.shape[1])
 
 
+@testing.requires_testing_data
+def test_append_channels():
+    """Test raw splitting / re-appending channel types
+    """
+    raw = Raw(fif_fname, preload=True)
+    raw_nopre = Raw(fif_fname, preload=False)
+    raw_eeg = raw.pick_types(meg=False, eeg=True, copy=True)
+    raw_meg = raw.pick_types(meg=True, eeg=False, copy=True)
```

Comment:
you could also do `raw_stim = raw.pick_types(meg=False, eeg=False, stim=True, copy=True)` and check that none of those channels are in your `raw_new`